temporalio/activity.py

```diff
@@ -221,6 +231,36 @@ def is_cancelled() -> bool:
     return _Context.current().cancelled_event.is_set()
 
 
+@contextmanager
+def shield_thread_cancel_exception() -> Iterator[None]:
+    """Context manager for synchronous multithreaded activities to delay
+    cancellation exceptions.
+
+    By default, heartbeating synchronous multithreaded activities have an
```

Comment:
You should mention local activities don’t have to heartbeat to be cancelled here and in the readme

Reply:
Will do

Reply:
Done